storage_controller/src/id_lock_map.rs

```diff
@@ -52,3 +94,89 @@ where
         }
     }
 }
+
+pub async fn trace_exclusive_lock<
```

Comment:
I am not sure what would be the best way to test these two trace functions... 

I have looked into [tracing_test](https://docs.rs/tracing-test/latest/tracing_test/) but it seems just a bit too heavy to use in unit tests and slows it down for the `LOCK_TIMEOUT_ALERT_THRESHOLD` duration. 
On the other hand, I see that there are logging tests in `test_runner/regress/test_logging.py` but here I am not sure how would I cause these locks to hold for the duration to escalate to the point of being logged consistently.

Reply:
How about writing a python regress test. You could:
1. Add a `pausable_failpoint` to one of the exclusive operations done by the storage controller
2. Configure the failpoint
3. Send an HTTP request which triggers the failpoint
4. Send another HTTP request which triggers an attempt to grab the lock
5. At the end of the test validate that the warning were printed

`NeonStorageController` python class will be useful. Also, the tests in `test_storage_controller.py` will be useful as examples. The `NeonPageserver` python class already has a `log_contains` and you'll need to add a similar one to `NeonStorageController`.

I'm concious this is rather involved, but such a test would be very valuable. Let me know if you need help (e.g. running the tests locally).